pynamodb/connection/base.py

```diff
@@ -254,8 +255,9 @@ def _make_api_call(self, operation_name, operation_kwargs):
         )
         prepared_request = self.client._endpoint.create_request(request_dict, operation_model)
 
-        for attempt_number in range(1, self._max_retry_attempts_exception + 1):
-            is_last_attempt_for_exceptions = attempt_number == self._max_retry_attempts_exception
+        for i in range(0, self._max_retry_attempts_exception + 1):
+            attempt_number = i + 1
```

Comment:
The attempt_number is used for logging at multiple places. If I remove it, I would have to do i+1, everywhere. Not so clean/manageable/uniform